swdb_2019_tools/Neuropixels_timeseries.py

```diff
@@ -0,0 +1,188 @@
+#Function that takes in: Session ID(s) as list (sessionIDs),
+    #brain regions (using manual_structure_acronym) as list (regions),
+    #and datatype as 'spikes', 'lfp', or 'both' (defaults to both)
+    #IF session IDs is left blank, then it will default to search through all the sessions that contain the brain regions(s)
+    #and will grab all the data across all sessions in which the brain region(s) was recorded
+#Spits out a PANDAS DATAFRAME for each datatype- if asking for "both", then provide a variable name for EACH dataframe:
+    #LFP contains:
+        #one row for each channelID,
+        #the timeseries array as a list,
+        #the channel's vertical, horizontal, and structure label,
+        #the probe ID that the channel belongs to,
+        #the mouse genotype,
+        #and the session ID
+    #Spikes contains:
+        #one row for each unit ID,
+        #unit's timeseries array as a list,
+        #the channel on which the unit was recorded,
+        #the channel's vertical, horizontal, and structure label,
+        #the probe ID that the channel belongs to,
+        #the mouse genotype,
+        #and the session ID
+#Import all the things
+import os
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+from collections import defaultdict
+
+from allensdk.brain_observatory.ecephys.ecephys_project_cache import EcephysProjectCache
+from allensdk.brain_observatory.ecephys import ecephys_session
+%matplotlib inline
+
+# fix slow autocomplete
+%config Completer.use_jedi = False
+
+#Set path to cache
+import platform
+platstring = platform.platform()
+
+if 'Darwin' in platstring:
+    # OS X
+    data_root = "/Volumes/Brain2019/"
+elif 'Windows'  in platstring:
+    # Windows (replace with the drive letter of USB drive)
+    data_root = "E:/"
+elif ('amzn1' in platstring):
+    # then on AWS
+    data_root = "/data/"
+else:
+    # then your own linux platform
+    # EDIT location where you mounted hard drive
+    data_root = "/media/$USERNAME/Brain2019/"
+
+manifest_path = os.path.join(data_root, "dynamic-brain-workshop/visual_coding_neuropixels/2019/manifest.json")
+
+#Get cache and info about all the sessions
+cache = EcephysProjectCache.fixed(manifest=manifest_path)
+sessions = cache.get_sessions()
+allchannelsinfo = cache.get_channels()
+allunitsinfo = cache.get_units()
+
+
+def get_all_timeseries_to_df(sessionIDs = [], regions = [], datatype = "both"):
```

Comment:
I think you want to delete all the things here that aren't in your function.. having the things outside the function will cause them to be run when someone imports this file.